clip_video_encode/clip_video_encode.py

```diff
@@ -120,36 +183,152 @@ def clip_video_encode(
+                shard_id = shard.split("/")[-1]
+                writer.create_shard(shard_id=int(shard_id.split(".tar")[0]))
+                with tarfile.open(tempdir + "/" + shard_id) as tar:
+                    tar.extractall(tempdir)
+                times["download_and_extract"] = times.get("download_and_extract", 0) + time.time() - t
+                t = time.time()
+                vids, ids, meta = read_shard(tempdir)
+                meta_refs = list(range(len(vids)))
+                fr = FrameReader(
+                    vids,
+                    meta_refs,
+                    take_every_nth,
+                    IMG_SIZE,
+                    workers=frame_workers,
+                    memory_size=frame_memory_size,
+                )
+                fr.start_reading()
 
-    if len(frames) > 0:  # TODO: make this cleaner
-        encode_chunk(frames, ind_dict, writer, fm, preprocess, meta, ids, use_dst_name, device)
+                frames, ind_dict = [], {}
+                block_size = 0
+                i = 0
+                n_frames = 0
+                for vid_frames, info in fr:
+                    i += 1
+                    n_frames += len(vid_frames)
+                    frames.append(vid_frames)
+                    ind_dict[info["reference"]] = (
+                        block_size,
+                        block_size + vid_frames.shape[0],
+                        info["dst_name"],
+                    )
+                    block_size += vid_frames.shape[0]
+                    times["read_frames"] = times.get("read_frames", 0) + time.time() - t
+                    t = time.time()
+
+                    if i % CHUNK_SIZE == 0:
+                        encode_chunk(
+                            frames,
+                            ind_dict,
+                            writer,
+                            fm,
+                            preprocess,
+                            meta,
+                            ids,
+                            use_dst_name,
+                            device,
+                        )
+                        times["encode"] = times.get("encode", 0) + time.time() - t
+                        t = time.time()
+                        frames, ind_dict, block_size = [], {}, 0
+                t = time.time()
+                if len(frames) > 0:  # TODO: make this cleaner
+                    encode_chunk(
+                        frames,
+                        ind_dict,
+                        writer,
+                        fm,
+                        preprocess,
+                        meta,
+                        ids,
+                        use_dst_name,
+                        device,
+                    )
+                times["encode"] = times.get("encode", 0) + time.time() - t
+                t = time.time()
+            finally:
+                shutil.rmtree(tempdir)
+                # writer.close()
+                print(f"Frames: {n_frames}")
```

Comment:
Maybe clean this up

Reply:
or just make it more simple - one line with the most important performance information